samples_project/Assets/SampleViewer/Samples/Measure/Measure.cs

```diff
@@ -14,272 +14,223 @@
+
+				lineMarker.GetComponent<ArcGISLocationComponent>().enabled = true;
+				lineMarker.GetComponent<ArcGISLocationComponent>().Position = thisPoint;
+				lineMarker.GetComponent<ArcGISLocationComponent>().Rotation = new ArcGISRotation(0, 90, 0);
+
+				if (stops.Count > 0)
+				{
+					GameObject lastStop = stops.Peek();
+					var lastPoint = lastStop.GetComponent<ArcGISLocationComponent>().Position;
+
+					//calculate distance from last point to this point
+					geodeticDistance += ArcGISGeometryEngine.DistanceGeodetic(lastPoint, thisPoint, new ArcGISLinearUnit(ArcGISLinearUnitId.Meters), new ArcGISAngularUnit(ArcGISAngularUnitId.Degrees), ArcGISGeodeticCurveType.Geodesic).Distance;
+					GeodeticDistanceText.text = "Distance: " + Math.Round(geodeticDistance, 3).ToString() + unitTxt;
+
+					featurePoints.Add(lastStop);
+					//interpolate middle points between last point and this point
+					Interpolate(lastStop, lineMarker, featurePoints);
+					featurePoints.Add(lineMarker);
+
+				}
+				//add this point to stops and also to feature points where stop is user-drawed, and feature points is a collection of user-drawed and interpolated
+				stops.Push(lineMarker);
+				RenderLine(ref featurePoints);
+				RebaseLine();
+
+			}
+
+		}
+	}
+
+	private void Interpolate(GameObject start, GameObject end, List<GameObject> featurePoints)
+	{
+		var startPoint = start.GetComponent<ArcGISLocationComponent>().Position;
+		var endPoint = end.GetComponent<ArcGISLocationComponent>().Position;
+
+		double d = ArcGISGeometryEngine.DistanceGeodetic(startPoint, endPoint, new ArcGISLinearUnit(ArcGISLinearUnitId.Meters), new ArcGISAngularUnit(ArcGISAngularUnitId.Degrees), ArcGISGeodeticCurveType.Geodesic).Distance;
+		float n = Mathf.Floor((float)d / InterpolationInterval);
+		double dx = (end.transform.position.x - start.transform.position.x) / n;
+		double dz = (end.transform.position.z - start.transform.position.z) / n;
+
+		var pre = start.transform.position;
+
+		//calculate n-1 intepolation points/n-1 segments because the last segment is already created by the end point 
+		for (int i = 0; i < n - 1; i++)
+		{
+			GameObject next = Instantiate(InterpolationMarker, arcGISMapComponent.transform);
+
+			//calculate transform of next point
+			float nextX = pre.x + (float)dx;
+			float nextZ = pre.z + (float)dz;
+			next.transform.position = new Vector3(nextX, 0, nextZ);
+
+			//set default location component of next point
+			next.GetComponent<ArcGISLocationComponent>().enabled = true;
+			next.GetComponent<ArcGISLocationComponent>().Rotation = new ArcGISRotation(0, 90, 0);
+
+			//define height
+			SetElevation(next);
+
+			featurePoints.Add(next);
+
+			pre = next.transform.position;
+		}
+
+	}
+
+	// set height for point transform and location component
+	void SetElevation(GameObject stop)
+	{
+		// start the raycast in the air at an arbitrary to ensure it is above the ground
+		var raycastHeight = 5000;
```

Comment:
Maybe make this a constant variable. This measure code wont work in areas above 5000 elevation.